### PR TITLE
interfaces,overlorf/ifacestate: honor user choice and don't auto-connect disconnected plugs

### PR DESCRIPTION
**Pull request by zyga**

### CODE REVIEW

**Comment by zyga** [comment]
retest this please

File: interfaces/repo.go [hunk 1]
```diff
@@ -655,6 +655,32 @@ func (r *Repository) RemoveSnap(snapName string) error {
 	return nil
 }
 
+// AutoConnectBlacklist returns plugs that should not be auto-connected.
```
Comment (niemeyer) [inline interfaces/repo.go]: s/plugs/plug names/

File: interfaces/repo.go [hunk 2]
```diff
@@ -655,6 +655,32 @@ func (r *Repository) RemoveSnap(snapName string) error {
...
+//
+// Plug is blacklisted if it has no connections despite using an auto-connected
+// interface. That implies it was manually disconnected after installation.
```
Comment (niemeyer) [inline interfaces/repo.go]: s/after installation//

We'll likely support blacklisting _during_ installation at some point.

**Comment by niemeyer** [comment]
LGTM

**Pull request merged**